src/app/Dashboard/Charts/ChartCard.tsx

```diff
@@ -0,0 +1,360 @@
+          isDisabled={!chartSrc || !dashboardUrl}
+        />
+      </>
+    );
+  }, [props.chartKind, updateKey, chartSrc, dashboardUrl]);
+
+  const popoutButton = React.useMemo(() => {
+    return (
+      <>
+        <Button
+          key={1}
+          aria-label={`Pop out ${props.chartKind} chart`}
+          onClick={popout}
+          variant="plain"
+          icon={<ExternalLinkAltIcon />}
+          isDisabled={!chartSrc || !dashboardUrl}
+        />
+      </>
+    );
+  }, [props.chartKind, popout, chartSrc, dashboardUrl]);
+
+  const actions = React.useMemo(() => {
+    const a = props.actions || [];
+    return [expandButton, popoutButton, ...a];
+  }, [props.actions, expandButton, popoutButton]);
+
+  const header = React.useMemo(() => {
+    const isWide = props.span > 4;
+    const style = {
+      marginBottom: isWide ? '-2.8em' : '',
+    };
+    if (hasRecording) {
+      return (
+        <>
+          <CardHeader style={style}>
+            <CardActions>{actions}</CardActions>
+          </CardHeader>
+        </>
+      );
+    } else {
+      return (
+        <>
+          <CardHeader style={style}>
+            <CardTitle>{props.chartKind}</CardTitle>
+            <CardActions>{props.actions}</CardActions>
+          </CardHeader>
+        </>
+      );
+    }
+  }, [props.actions, props.span, props.chartKind, hasRecording, actions]);
+
+  const handleCreateRecording = React.useCallback(() => {
+    history.push({
+      pathname: '/recordings/create',
+      state: {
+        restartExisting: true,
+        name: RECORDING_NAME,
+        templateName: 'Profiling',
+        templateType: 'TARGET',
+        labels: [{ key: 'origin', value: RECORDING_NAME }],
+        duration: -1,
+        // TODO these are arbitrary defaults that will be set in the recording creation form.
+        // Should these values be inferred in some more intelligent way?
+        maxAge: 120, // seconds
+        maxSize: 100 * 1024 * 1024, // bytes
+      } as CreateRecordingProps,
+    });
+  }, [history]);
+
+  return (
+    <>
+      <DashboardCard
+        key={idx}
+        id={props.chartKind + '-chart-card'}
+        dashboardId={props.dashboardId}
+        cardSizes={ChartCardSizes}
+        isCompact
+        style={cardStyle}
+        cardHeader={header}
+      >
+        <CardBody>
+          {hasRecording ? (
+            <iframe key={key} style={{ height: '100%', width: '100%' }} src={chartSrc} />
+          ) : (
+            <Bullseye>
+              <EmptyState variant={EmptyStateVariant.large}>
+                <EmptyStateIcon icon={DataSourceIcon} />
+                <Title headingLevel="h2" size="md">
+                  No source recording
+                </Title>
+                <EmptyStateBody>
+                  Metrics cards display data taken from running flight recordings with the label{' '}
+                  <code>origin={RECORDING_NAME}</code>. No such recordings are currently available.
```

Comment:
We can replace `<code />` with PF label here.

```typscript
<Label color='blue' isCompact>origin={RECORDING_NAME}</Label>
```

![Screenshot from 2023-02-09 01-08-49](https://user-images.githubusercontent.com/68053619/217732135-b67e8d48-3769-45b6-889a-3b19752f87c0.png)